client/core/trade.go

```diff
@@ -1659,18 +1629,26 @@ func (c *Core) redeemMatchGroup(t *trackedTrade, matches []*matchTracker, errs *
 		proof := &match.MetaData.Proof
 		coinID := []byte(coinIDs[i])
 		if match.Match.Side == order.Taker {
-			match.SetStatus(order.MatchComplete) // could this cause the match to be retired before the `redeem` request succeeds?
+			match.SetStatus(order.MatchComplete)
+			// match.MetaData.Retired = true // Don't retire until `redeem` request to DEX completes.
 			proof.TakerRedeem = coinID
 		} else {
 			match.SetStatus(order.MakerRedeemed)
 			proof.MakerRedeem = coinID
 		}
+		revoked := match.IsRevoked()
+		if revoked {
+			match.MetaData.Retired = true // Redemption or refund is the last step for revoked matches.
+		}
```

Comment:
Pls comment that this would be expected in the auto-redeem situation where counterparty broadcasted their redeem too late and without server coordination.

But this an example of where we've distributed the responsibility of deciding "activeness" from one place to all over core, and I think that's more error prone than doing/checking with a single piece of code.